### ISSUE TITLE
Destroying the LLVM Module is slow

### ISSUE THREAD

**Issue by pcwalton** [rust-lang/rust#3552]
About 3.1% of rustc compilation time (2.6 s) is spent destroying the LLVM module. For comparison, that's longer than all of resolve.

The solution is to leak the module and let the OS clean up all the memory.

**Comment by steveklabnik** [comment]
Triage: nothing has moved on this.

**Comment by BigBoneDaddy** [comment]
@steveklabnik Still with out movement?

**Comment by brson** [comment]
Nobody has looked at this in a long time. We can quickly test whether this is still a win: just remove all calls to `llvm::LLVMDisposeModule(llmod);` and compare the times of a clean rustc build before and after. If it improves compile time then we can make it an option with a flag in e.g. Session and turn it on for rustc.

If it doesn't improve compile time then we can finally close this issue.

**Comment by brson** [comment]
On the other hand, this bug is based on four-year old information. Maybe we should just let it go.

**Comment by sanxiyn** [comment]
Updated numbers on librustc: "name resolution" takes 0.3 seconds, "LLVM passes" takes 14.4 seconds, "LLVM passes" without dispose takes 14.1 seconds. So it is still true LLVM dispose takes as long as resolve.

On the other hand, total time is 50 seconds, so both LLVM dispose and resolve takes less than 1% of total time. I think we can close this.

Note: `llvm::LLVMContextDispose(llcx)` also needs to be removed, because context destructor calls module destructor. When I didn't remove it, memory usage in `-Z time-passes` stayed same. When I did, memory usage post-LLVM increased by 400 MB.
